### DISCUSSION TITLE
Pixela max number ?

### DISCUSSION THREAD

**Discussion by ddijanic** [forkineye/ESPixelStick#400] [Q&A]
written 800 pixels is 25 ms .. that's for eps8266 or eps32. because it is the maximum number of pixels per device or per ip out (port)?
just table for eps32 and esp8266 how many pixels per port at how many fps would be useful.

**Comment by MartinMueller2003** [comment]
The code specifies total Pixels per ESP. The ESP8266 supports a total of 800 pixels and there is only a single output port. The ESP32 version supports 3000 pixels spread across all active output ports.

NOTE: The input and output refresh rates are independent of each other. The rate at which data is delivered to the ESP does not control the rate at which a pixel string gets updated.

How many pixels you configure per output port will determine the output refresh rate. The larger the pixel count on a given port, the slower the refresh rate. The smaller the pixel count on a given port, the faster the refresh rate.

To get 25ms or faster refresh rates using WS281x pixels. you need to have 680 or fewer pixels in a pixel string.

FYI: 800 pixels in a WS281x string will get you a ~30ms refresh rate.

**Comment by Mat-Moo** [comment]
I'm about to break my esp286 with a new prop so need to go esp32! It should work with any generic esp32?

**Comment by MartinMueller2003** [comment]
I use LoLin D32 Pro. It has the largest number of available ports. We also support an image for the ESP32-cam boards which have fewer possible output ports.

**Reply by Mat-Moo** [reply]
https://www.amazon.co.uk/gp/aw/d/B0814WGPVY/ref=ox_sc_act_image_1?smid=A1D4HALQO7H40T&psc=1
D
Should be ok? Would like just to test for now, only looking to drive 1200 pixels

**Comment by forkineye** [comment]
Moving to discussions

**Comment by Mat-Moo** [comment]
Not sure on the terminology, I want to run a grid 24x48 (1152 Pixels) over a ESP32 - I've got 4 beta 4 firmware on here and seems to be running ok. Got static IP and set primary input as E1.31 and two outputs 1 and 2 - What is the difference between start universe and First channel in universe? How do I set the second output to play the second part of the E1.31 stream? Start null count? End null count? What do these mean? @MartinMueller2003 any help welcome!

**Reply by forkineye** [reply]
The start channel in the universe is an offset into that universe to start listening for data. The null counts are pixels in your string that you always want to be rendered blank. For instance, if you have a string of 100 pixels but are only using 95 of them, you could either cut the string or set the null end count to 5.

**Reply by Mat-Moo** [reply]
Ok, null pixels I understand, not using :)

But, still unsure on setup for grid. I split my grid into 2 of 24x24 (576 pixels) running off two channels (on the ESP32). In Vixen I've set that as one output with a static IP address and as a 24 x 48 grid, 1152 outputs (3456 bytes). 

Does the v4 stream the first 576 pixels to output channel 1 then the second 576 to output channel 2, or do I need to split that up at software layer to 2 grids not one, and configure them as different universes etc.? 

Sorry I've not got all the hardware connected to test with fully still at this stage, so that makes it harder for me as well :(

**Comment by MartinMueller2003** [comment]
You are on the right track. The ESP takes the first X channels and sends them to port 1. It then sends Y channels to port two and Z channels to port three etc. You define X Y Z on the configuration page. I say it this way because the ESP has no idea how you set up the show player and does not have any way to know your channel counts. So you set up port 1 to send 576 pixels and then you configure port two to send 576 pixels and you configure the show player to send the ESP 1152 pixels.

**Reply by Mat-Moo** [reply]
Thanks, I'm so close but getting a huge amount of flashing only on the 2nd half of the grid. I thought it was because of the distance from the esp 2 the 2nd part of the grid, but even with a level shifter I still get flashes. 
Wondering what to do next, thinking of trying another pin?

**Comment by Mat-Moo** [comment]
I’ve abandoned 2 channels, and moved to run on one port with all 1152 nodes, and works perfectly it seems? But I cannot get this to work on second channel without flickering ;( also when using the rainbow test patter I notice odd red flashing pixels, but don’t get that through vixen etc. So maybe something odd?

**Reply by Mat-Moo** [reply]
Sorry terminology, I've abandoned 2 port output, still sending as one channel from vixen (well however many channels it requires) will mean slower but not flashing until I can work out a better plan

**Comment by henrygab** [comment]
> ... I cannot get this to work on second channel without flickering ;( also when using the rainbow test patter I notice odd red flashing pixels ...

When I first setup a matrix, I had similar problems. Eventually, it turns out my problem was not related to the voltage of the data line, but with the ground. If you add an extra, oversized, ground wire, and connect it to the second strip in multiple spots, does it help?

_I'm presuming you have a nice beefy capacitor at the start of the second segment._

**Comment by Mat-Moo** [comment]
Ok, so today I've rewired, and added another ground from esp32 to second string on the grid. Two outputs 576 each, and not seeing a lot of flashing now so looks a bit better... but it's just not smooth scrolling just seems jerky and depressing :( any clues? this is still running 4 beta 4 - worth a recompile of newer version? (using pins 2 and 14 and board looks like https://lastminuteengineers.com/esp32-pinout-reference/) so both on SPI pins

**Reply by MartinMueller2003** [reply]
What is your refresh rate and is it jerky when you use the built in effects?

**Reply by Mat-Moo** [reply]
Oh bum I just plumbed in a esp32cam with wled to test - silky smooth :( the difference is chalk and cheese, I've had this grid for 6 months and almost gave up, thought I'd play again today, rewired, all excited, same deal. Maybe it's the esp32 I'm using, I had a second esp32cam but I can't find the bugger anywhere. Maybe I just wait for the next release and try again, but for now Wled does a better job :(

**Reply by Mat-Moo** [reply]
also noted https://github.com/forkineye/ESPixelStick/pull/473 would that help?
